sdk/core/src/etag.rs

```diff
@@ -0,0 +1,19 @@
+use std::fmt;
+
+#[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize)]
+pub struct Etag(String);
+
+impl<T> From<T> for Etag
+where
+    T: Into<String>,
+{
+    fn from(t: T) -> Self {
+        Self(t.into())
+    }
+}
+
+impl fmt::Display for Etag {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
```

Comment:
For now this may be fine, but we need to support weak etags at some point. See https://azure.github.io/azure-sdk/dotnet_introduction.html#conditional-request-methods for some links. We do have sevices that depend on this (e.g. AppConfig).